sdk/storage/Azure.Storage.DataMovement/src/ServiceToServiceJobPart.cs

```diff
@@ -414,5 +420,19 @@ internal async Task DisposeHandlers()
                 await _commitBlockHandler.DisposeAsync().ConfigureAwait(false);
             }
         }
+
+        private StorageResourceCopyFromUriOptions GetCopyFromUriOptions()
```

Comment:
Why does the base package need to know about auth scheme at all? Can't the implementation of the resource just know what auth it should use and just do it rather than the base class telling it? Maybe I'm missing something here.